x-pack/plugin/core/src/main/java/org/elasticsearch/index/engine/frozen/FrozenEngine.java

```diff
@@ -133,14 +140,14 @@ public CacheHelper getReaderCacheHelper() {
     }
 
     @SuppressForbidden(reason = "we manage references explicitly here")
-    private synchronized void onReaderClosed(IndexReader.CacheKey key) {
+    private synchronized void onReaderClosed(ElasticsearchDirectoryReader reader) {
         // it might look awkward that we have to check here if the keys match but if we concurrently
         // access the lastOpenedReader there might be 2 threads competing for the cached reference in
         // a way that thread 1 counts down the lastOpenedReader reference and before thread 1 can execute
         // the close listener we already open and assign a new reader to lastOpenedReader. In this case
         // the cache key doesn't match and we just ignore it since we use this method only to null out the
```

Comment:
can you fix this comment to no longer reference the cache key since we check the readers directly?

Reply:
revert in f891f88ff1d